src/Microsoft.AspNet.Hosting.Abstractions/EnvironmentName.cs

```diff
@@ -0,0 +1,11 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+namespace Microsoft.AspNet.Hosting
+{
+    public static class EnvironmentName
```

Comment:
I think singular is maybe more correct, no? Think of this as a "Mode." Would it be `AppMode { ModeA, ModeB, ModeC }` or `AppModes { ModeA, ModeB, ModeC }`? When you set it, is it `CurrentMode = AppMode.ModeA;` or `CurrentMode = AppModes.ModeA;`